src/nylas.ts

```diff
@@ -130,38 +132,36 @@ class Nylas {
       throw new Error('exchangeCodeForToken() must be called with a code');
     }
 
-    return new Promise<string>((resolve, reject) => {
-      const options = {
-        method: 'GET',
-        json: true,
-        url: `${this.apiServer}/oauth/token`,
-        qs: {
-          client_id: this.clientId,
-          client_secret: this.clientSecret,
-          grant_type: 'authorization_code',
-          code: code,
-        },
-      };
-
-      return request(options, (error, response, body) => {
-        if (!body || !body['access_token']) {
-          let errorMessage = 'No access token in response';
-          if (body && body.message) errorMessage = body.message;
-          error = error ? error : new Error(errorMessage);
+    const url = new URL(`${this.apiServer}/oauth/token`);
+    url.searchParams.set('client_id', this.clientId);
+    url.searchParams.set('client_secret', this.clientSecret);
+    url.searchParams.set('grant_type', 'authorization_code');
+    url.searchParams.set('code', code);
```

Comment:
using URL is an interesting choice here ;) is it just to urlencode querystring? looks good to me, just wanted to point out

Reply:
Browser support makes it more preferred choice over `querystring`